app/scripts/views/sign_up.js

```diff
@@ -197,7 +198,23 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
     },
 
     suggestEmail: function () {
-      mailcheck(this.$el.find('.email'), this.metrics, this.translator, this.window.location.search);
+      var abChoose;
+      var abData = {
+        isMetrics: !! this.metrics._able,
+        uuid: this.metrics._uuid
+      };
+
+      var mailcheckParamChoice = Url.searchParam('mailcheck', this.window.location.search);
```

Comment:
Maybe most of https://github.com/mozilla/fxa-content-server/blob/faa0a02d459b7c6c67bc774295700422a1cb3882/app/scripts/models/mixins/search-param.js#L56 should actually live in url.js and used here?